### PR TITLE
GH-4014: MQTT ClientManager: completion timeouts

### LINKED ISSUE #4014: Reusing of MQTT connection (clientId) results in error log "... Never happens."
This issue is slightly realated to GH-3685 (https://github.com/spring-projects/spring-integration/issues/3685).

I am trying to use one MQTT connection with a single clientId for subscriber as well as publisher channels.

Problem is that in <tt>MqttConnectOptions</tt> is the timeout specified in *seconds* while in <tt>org.eclipse.paho.client.mqttv3.MqttToken.waitForCompletion(long)</tt> is expected in *milliseconds*.

<pre>
MqttConnectOptions.setConnectionTimeout(int seconds)
client.connect(this.connectionOptions).waitForCompletion(long milliseconds)
</pre>

So calling:

<pre>
Mqttv3ClientManager.start() {
  ...
  client.connect(this.connectionOptions).waitForCompletion(this.connectionOptions.getConnectionTimeout());
  ...
}
</pre>

times out very quickly (default seems to be 30 seconds but they actually time out in 30 milliseconds) and an attempt to call the <tt>reconnect()</tt> method is made even though the connection is still becoming connected. That results in:

<pre>
17:17:04,334 ERROR [main] o.s.i.m.c.Mqttv3ClientManager:110 MQTT client failed to connect. Never happens.
Connect already in progress (32110)
        at org.eclipse.paho.client.mqttv3.MqttAsyncClient.reconnect(MqttAsyncClient.java:1399)
        at org.springframework.integration.mqtt.core.Mqttv3ClientManager.start(Mqttv3ClientManager.java:107)
        at org.springframework.context.support.DefaultLifecycleProcessor.doStart(DefaultLifecycleProcessor.java:178)
</pre>

This just prints the error message in log, but the connection seems to be established without further problems and the application works.

### LINKED ISSUE #8550: MQTT is not re-subscribing after reconnect
**In what version(s) of Spring Integration are you seeing this issue?**

**6.0.2 with Spring Boot 3.0.2**

**Describe the bug**

When the client loses connection (and auto-reconnect is on) and then reconnects, existing subscriptions are not re-subscribbed. Application restart is needed to re-subscribe.

**To Reproduce**

- Start the sample application
- Connect to the MQTT broker
- Send message
- Stop the MQTT broker
- Start the MQTT broker
- Send message

After the reconnection, no message will appear in the sample application.

**Expected behavior**

I expect to re-subscribe to the topic and receive messages

**Sample**

The sample includes MQTT broker in the `infra` folder.

[Sample Here](https://github.com/iamceph/spring-integration-mqtt-bug).

[Original thread on StackOverflow](https://stackoverflow.com/questions/75404056/spring-integration-mqtt-not-re-subscribing-after-reconnect)

### PR DESCRIPTION
**Pull request by artembilan**
Fixes https://github.com/spring-projects/spring-integration/issues/4014

The `ClientManager` implementations uses by mistake a `connectionTimeout` for operations with completion wait

* Introduce `completionTimeout` and `disconnectCompletionTimeout` for `ClientManager` abstraction to realign the logic with existing channel adapters and Paho Client by itself.
* Deprecate `DEFAULT_COMPLETION_TIMEOUT` and `DISCONNECT_COMPLETION_TIMEOUT` constants in the `AbstractMqttMessageDrivenChannelAdapter` in favor of respective replacement in the `ClientManager`
* Pull `disconnectCompletionTimeout` property from the `MqttPahoMessageDrivenChannelAdapter` to its superclass
* Use new `disconnectCompletionTimeout` in the `Mqttv5PahoMessageDrivenChannelAdapter` for similar `disconnectForcibly()` call
* Fix Lifecycle race condition when `ClientManager` is started by the outbound channel adapter (`Integer.MIN_VALUE` phase and auto-startup - see `DefaultLifecycleProcessor.doStart()` and logic around `dependencies`) which is much earlier than `MessageProducerSupport` (`Integer.MAX_VALUE / 2` phase) and there a `connectComplete()` callback might be called before the `MessageProducerSupport.start()`. For that purpose check for an `isRunning()` in the `connectComplete()` before subscribing and set `readyToSubscribeOnStart` flag to `subscribe()` in a `doStart()` of this `MqttMessageDrivenChannelAdapter`
* Remove redundant `MqttPahoMessageDrivenChannelAdapter.cleanSession` property in favor of `this.clientFactory.getConnectionOptions().isCleanSession()` call

GH-8550: MQTT: Always re-subscribe on re-connect

Fixes https://github.com/spring-projects/spring-integration/issues/8550

Turns out the Paho MQTT client does not re-subscribe when connection re-established on automatic reconnection

* Fix `AbstractMqttMessageDrivenChannelAdapter` to always subscribe to their topics in the `connectComplete()` independently of the `reconnect` status
* Verify behavior with `MOSQUITTO_CONTAINER` image restart in Docker

### CODE REVIEW

**Comment by artembilan** [comment]
This PR also includes the fix for https://github.com/spring-projects/spring-integration/issues/8550.
Otherwise an existing PR for that one is in conflict.

File: spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/core/AbstractMqttClientManager.java [hunk 1]
```diff
@@ -53,6 +53,10 @@
 
 	private int phase = DEFAULT_MANAGER_PHASE;
 
+	private long completionTimeout = ClientManager.DEFAULT_COMPLETION_TIMEOUT;
```
Comment (garyrussell) [inline spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/core/AbstractMqttClientManager.java:56]: Consider using the more modern `Duration`?

**Review by garyrussell — COMMENTED** [review]
Just one comment/suggestion. Will merge as-is if you prefer.

File: spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/core/AbstractMqttClientManager.java (hunk 1 above)
Comment (artembilan) [reply spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/core/AbstractMqttClientManager.java:56]: I will, when the target client lib does, but now for consistency I would stay with the same type the target API expects from us.

**Review by garyrussell — APPROVED** [review]

**Pull request merged**